src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -24,10 +27,12 @@ public class TemplateAnalyzer
         /// </summary>
         /// <param name="template">The ARM Template <c>JSON</c>. Must follow this schema: https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#</param>
         /// <param name="parameters">The parameters for the ARM Template <c>JSON</c></param>
-        public TemplateAnalyzer(string template, string parameters = null)
+        /// <param name="templateFilePath">The ARM Template file path. Needed to run arm-ttk checks.</param> TODO improve TemplateAnalyzer interface
```

Comment:
No, I wanted to discuss that here in the PR, thank you for bringing it up :)

I know having to pass the template andd the file path as arguments is not ideal but as I mentioned in your other comment I don't see another way around it

What I would want to do at least is to change the interface of TemplateAnalyzer by removing the constructor and only passing the needed parameters in EvaluateRulesAgainstTemplate, as that is the only public-facing function that actually uses that data. But I wanted to check with you first, because I noticed constructors are used like this also in other classes, and I was wondering if there's a reason for that